crates/provider/src/provider/trait.rs

```diff
@@ -134,26 +137,58 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
     /// # let tx = alloy_rpc_types_eth::transaction::TransactionRequest::default();
     /// // Execute a call on the latest block, with no state overrides
     /// let output = provider.call(&tx).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "eth_call")]
+    #[doc(alias = "call_with_overrides")]
+    fn call<'req>(
+        &'req self,
+        tx: &'req N::TransactionRequest,
+    ) -> ProviderCall<T, EthCallParams<'req, 'req, N>, Bytes, Bytes> {
+        let call = self.call_internal(tx);
+        call.into_provider_call()
+    }
```

Comment:
this is a breaking api change, that forces usage of call_internal if you want to use EthCall functions. I don't think this is what we want here, because this is very counter-intuitive.